### PR TITLE
test: fix tsan bug in integration test

### PR DESCRIPTION
**Pull request by akonradi**
Commit Message: Fix TSAN failure in overload_integration_test
Additional Description:
Avoid deleting the HTTP stream prematurely, and use a test fixture variable instead of a local one.

Risk Level: low
Testing: ran TSAN test 100 times
Docs Changes: n/a
Release Notes: n/a
Platform Specific Features: n/a

### CODE REVIEW

**Review by antoniovicente — APPROVED** [review]
Thanks for the fast turn around on a fix.  Waiting for CI before merging.

**Review by alyssawilk — APPROVED** [review]
Thanks for the quick fix!

**Pull request merged**
